lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -1866,26 +1866,31 @@ sub investigate ($self, %args) {
         next unless $prev->result =~ /(?:passed|softfailed)/;
         $inv{last_good} = {type => 'link', link => '/tests/' . $prev->id, text => $prev->id};
         last unless $prev->result_dir;
-        # just ignore any problems on generating the diff with eval, e.g.
-        # files missing. This is a best-effort approach.
-        my @files = map { Mojo::File->new($_->result_dir(), 'vars.json')->slurp } ($prev, $self);
-        my $diff = eval { diff(\$files[0], \$files[1], {CONTEXT => 0}) };
-        $inv{diff_to_last_good} = join("\n", grep { !/(^@@|$ignore)/ } split(/\n/, $diff));
-        $inv{diff_packages_to_last_good} = $self->packages_diff($prev, $ignore) // 'Diff of packages not available';
-        my ($before, $after) = map { decode_json($_) } @files;
-        my $dir = testcasedir($self->DISTRI, $self->VERSION);
-        my $refspec_range = "$before->{TEST_GIT_HASH}..$after->{TEST_GIT_HASH}";
-        $inv{test_log} = $self->git_log_diff($dir, $refspec_range);
-        $inv{test_log} ||= 'No test changes recorded, test regression unlikely';
-        $inv{test_diff_stat} = $self->git_diff($dir, $refspec_range) if $inv{test_log};
-        # no need for duplicating needles git log if the git repo is the same
-        # as for tests
-        if ($after->{TEST_GIT_HASH} ne $after->{NEEDLES_GIT_HASH}) {
-            $dir = needledir($self->DISTRI, $self->VERSION);
-            my $refspec_needles_range = "$before->{NEEDLES_GIT_HASH}..$after->{NEEDLES_GIT_HASH}";
-            $inv{needles_log} = $self->git_log_diff($dir, $refspec_needles_range);
-            $inv{needles_log} ||= 'No needle changes recorded, test regression due to needles unlikely';
-            $inv{needles_diff_stat} = $self->git_diff($dir, $refspec_needles_range) if $inv{needles_log};
+        my ($prev_file, $self_file) = map {
+            eval { Mojo::File->new($_->result_dir(), 'vars.json')->slurp }
+              // undef
+        } ($prev, $self);
+        if ($prev_file) {
```

Comment:
`last unless $prev_file;` would avoid having to indent all the code for this case.

Reply:
Also I think that at least the line `$inv{diff_packages_to_last_good} = $self->packages_diff($prev, $ignore) // 'Diff of packages not available';` is not relying on being able to read the files. I suggest to move that line up and then use

```suggestion
        last unless $prev_file;
```

Reply:
done